app/services/create_work_package_service.rb

```diff
@@ -0,0 +1,53 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+class CreateWorkPackageService
+  attr_accessor :user, :send_notifications, :project, :work_package
+
+  def initialize(user:, project:, send_notifications: true)
+    self.user = user
+    self.project = project
+    self.send_notifications = send_notifications
+
+    WorkPackageObserver.instance.send_notification = send_notifications
+  end
+
+  def create
+    hash = {
+      project: project,
+      author: user,
+      type: Type.where(is_default: true).first || Type.first
```

Comment:
Shouldn't the type take the project into account?

Currently it would be possible that this method chooses a type that is not part of the project.